tests/util/test_full_block_utils.py

```diff
@@ -243,3 +246,10 @@ async def test_parser(self):
             assert gen == block.transactions_generator
             # this doubles the run-time of this test, with questionable utility
             # assert gen == FullBlock.from_bytes(block_bytes).transactions_generator
+
+    @pytest.mark.asyncio
+    async def test_header_block(self):
+        for block in get_full_blocks():
+            hb: HeaderBlock = get_block_header(block, [], [])
+            hb_bytes = header_block_from_block(memoryview(bytes(block)))
+            assert HeaderBlock.from_bytes(hb_bytes) == hb
```

Comment:
I suppose this is a simple way of writing this test. but I would expect it to take a lot longer to run than it would need to. iirc, looping over all `get_full_blocks()` takes about 4 minutes. The `HeaderBlock` is smaller and I would expect there to be fewer variations of it to generate in the test.

Reply:
It takes about 90 seconds on my machine (I guess do to the serialization improvements from G1), I think it's not so bad.